Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# refer to the shape by its URI
app1_shape = URIRef("urn:simpleapp/app1")

print(sc.shape_to_query(app1_shape))
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[4], line 4[0m
[1;32m      1[0m [38;5;66;03m# refer to the shape by its URI[39;00m
[1;32m      2[0m app1_shape [38;5;241m=[39m URIRef([38;5;124m"[39m[38;5;124murn:simpleapp/app1[39m[38;5;124m"[39m)
[0;32m----> 4[0m [38;5;28mprint[39m([43msc[49m[38;5;241;43m.[39;49m[43mshape_to_query[49m[43m([49m[43mapp1_shape[49m[43m)[49m)

File [0;32m~/work/BuildingMOTIF/BuildingMOTIF/buildingmotif/dataclasses/shape_collection.py:312[0m, in [0;36mShapeCollection.shape_to_query[0;34m(self, shape)[0m
[1;32m    301[0m     [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21mshape_to_query[39m([38;5;28mself[39m, shape: URIRef) [38;5;241m-[39m[38;5;241m>[39m [38;5;28mstr[39m:
[1;32m    302[0m [38;5;250m        [39m[38;5;124;03m"""[39;00m
[1;32m    303[0m [38;5;124;03m        This method takes a URI representing a SHACL shape as an argument and returns[39;00m
[1;32m    304[0m [38;5;124;03m        a SPARQL query selecting the information which would be used to satisfy that[39;00m
[0;32m   (...)[0m
[1;32m    310[0m [38;5;124;03m            ?target <path> <value>[39;00m
[1;32m    311[0m [38;5;124;03m        """[39;00m
[0;32m--> 312[0m         clauses, project [38;5;241m=[39m [43m_shape_to_where[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mgraph[49m[43m,[49m[43m [49m[43mshape[49m[43m,[49m[43m [49m[38;5;124;43m"[39;49m[38;5;124;43m?target[39;49m[38;5;124;43m"[39;49m[43m)[49m
[1;32m    313[0m         preamble [38;5;241m=[39m [38;5;124m"""[39m[38;5;124mPREFIX sh: <http://www.w3.org/ns/shacl#>[39m
[1;32m    314[0m [38;5;124mPREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>[39m
[1;32m    315[0m [38;5;124mPREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>[39m
[1;32m    316[0m [38;5;124m        [39m[38;5;124m"""[39m
[1;32m    317[0m         [38;5;28;01mreturn[39;00m [38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00mpreamble[38;5;132;01m}[39;00m[38;5;124m SELECT [39m[38;5;132;01m{[39;00m[38;5;124m'[39m[38;5;124m [39m[38;5;124m'[39m[38;5;241m.[39mjoin(project)[38;5;132;01m}[39;00m[38;5;124m WHERE [39m[38;5;130;01m{{[39;00m[38;5;130;01m\n[39;00m[38;5;132;01m{[39;00mclauses[38;5;132;01m}[39;00m[38;5;130;01m\n[39;00m[38;5;130;01m}}[39;00m[38;5;124m"[39m

File [0;32m~/work/BuildingMOTIF/BuildingMOTIF/buildingmotif/dataclasses/shape_collection.py:412[0m, in [0;36m_shape_to_where[0;34m(graph, shape, root_var)[0m
[1;32m    410[0m qualified_pshapes: Set[Node] [38;5;241m=[39m [38;5;28mset[39m()
[1;32m    411[0m [38;5;28;01mfor[39;00m pshape [38;5;129;01min[39;00m graph[38;5;241m.[39mobjects(shape, SH[38;5;241m.[39mproperty):
[0;32m--> 412[0m     path [38;5;241m=[39m [43mshacl_path_to_sparql_path[49m[43m([49m[43mgraph[49m[43m,[49m[43m [49m[43mgraph[49m[38;5;241;43m.[39;49m[43mvalue[49m[43m([49m[43mpshape[49m[43m,[49m[43m [49m[43mSH[49m[38;5;241;43m.[39;49m[43mpath[49m[43m)[49m[43m)[49m
[1;32m    413[0m     [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m graph[38;5;241m.[39mvalue(pshape, SH[38;5;241m.[39mqualifiedValueShape):
[1;32m    414[0m         pshapes_by_path[path][38;5;241m.[39mappend(pshape)  [38;5;66;03m# type: ignore[39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/pyshacl/helper/path_helper.py:43[0m, in [0;36mshacl_path_to_sparql_path[0;34m(shapes_graph, path_node, prefixes, recursion)[0m
[1;32m     41[0m     [38;5;28;01mraise[39;00m ReportableRuntimeError([38;5;124m"[39m[38;5;124mPath traversal depth is too much![39m[38;5;124m"[39m)
[1;32m     42[0m top_level [38;5;241m=[39m recursion [38;5;241m==[39m [38;5;241m0[39m
[0;32m---> 43[0m sequence_list [38;5;241m=[39m [38;5;28mlist[39m([43mshapes_graph[49m[38;5;241;43m.[39;49m[43mgraph[49m[38;5;241m.[39mitems(path_node))
[1;32m     44[0m [38;5;28;01mif[39;00m [38;5;28mlen[39m(sequence_list) [38;5;241m>[39m [38;5;241m0[39m:
[1;32m     45[0m     all_collected [38;5;241m=[39m []

[0;31mAttributeError[0m: 'Graph' object has no attribute 'graph'

